tests/integration/dynamic_ip_test.py

```diff
@@ -671,72 +661,146 @@ def _get_nameservers():
     """
     Return a list of name server string configured in RESOLV_CONF_PATH.
     """
-    return (
+    running_ns = (
         libnmstate.show()
         .get(Constants.DNS, {})
         .get(DNS.RUNNING, {})
         .get(DNS.SERVER, [])
     )
+    logging.debug("Current running DNS: {}".format(running_ns))
+    return running_ns
 
 
 def _get_running_routes():
     """
     return a list of running routes
     """
-    return libnmstate.show().get(Constants.ROUTES, {}).get(RT.RUNNING, [])
-
-
-def _has_ipv6_auto_gateway():
-    routes = _get_running_routes()
-    for route in routes:
-        if (
-            route[RT.DESTINATION] == IPV6_DEFAULT_GATEWAY
-            and route[RT.NEXT_HOP_INTERFACE] == DHCP_CLI_NIC
-        ):
-            return True
-    return False
-
-
-def _has_ipv6_auto_extra_route():
-    routes = _get_running_routes()
-    for route in routes:
-        if (
-            route[RT.DESTINATION] == IPV6_CLASSLESS_ROUTE_DST_NET1
-            and route[RT.NEXT_HOP_INTERFACE] == DHCP_CLI_NIC
-        ):
-            return True
-    return False
-
-
-def _has_ipv6_auto_nameserver():
-    return DHCP_SRV_IP6 in _get_nameservers()
-
-
-def _has_ipv4_dhcp_nameserver():
-    return DHCP_SRV_IP4 in _get_nameservers()
-
-
-def _has_ipv4_dhcp_gateway():
-    routes = _get_running_routes()
-    for route in routes:
-        if (
-            route[RT.DESTINATION] == IPV4_DEFAULT_GATEWAY
-            and route[RT.NEXT_HOP_INTERFACE] == DHCP_CLI_NIC
-        ):
-            return True
-    return False
-
-
-def _has_ipv4_classless_route():
-    routes = _get_running_routes()
-    for route in routes:
-        if (
-            route[RT.DESTINATION] == IPV4_CLASSLESS_ROUTE_DST_NET1
-            and route[RT.NEXT_HOP_ADDRESS] == IPV4_CLASSLESS_ROUTE_NEXT_HOP1
-            and route[RT.NEXT_HOP_INTERFACE] == DHCP_CLI_NIC
-        ):
-            return True
-    return False
+    running_routes = (
+        libnmstate.show().get(Constants.ROUTES, {}).get(RT.RUNNING, [])
+    )
+    logging.debug("Current running routes: {}".format(running_routes))
+    return running_routes
+
+
+def _poll(func, timeout):
```

Comment:
IMHO it would be nicer to have the default here `timeout=20` and avoid the global `DEFAULT_TMO` constant, unless it is also used at some other place.

Reply:
I intend to not use magic number. That's why this constant got introduced.